### ISSUE TITLE
[Types] `doc.updateOne(doc)` gives types error in 6.0.15 and higher

### ISSUE THREAD

**Issue by hasezoey** [Automattic/mongoose#11101]
**Do you want to request a *feature* or report a *bug*?**
types

**What is the current behavior?**
in 6.0.15 a change was included that changed type `UpdateQuery` to use `DeepPartial` instead of `LeanDocument`, which now makes the error `Type 'unknown' is not assignable to type 'DeepPartial<unknown>'`

**If the current behavior is a bug, please provide the steps to reproduce.**
Full Error:

```txt
Argument of type 'Document<any, any, User> & User & { _id: ObjectId; }' is not assignable to parameter of type 'UpdateWithAggregationPipeline | UpdateQuery<Document<any, any, User> & User & { _id: ObjectId; }> | undefined'.
  Type 'Document<any, any, User> & User & { _id: ObjectId; }' is not assignable to type 'UpdateQuery<Document<any, any, User> & User & { _id: ObjectId; }>'.
    Type 'Document<any, any, User> & User & { _id: ObjectId; }' is not assignable to type 'ReadonlyPartial<_UpdateQueryDef<DeepPartial<Document<any, any, User> & User & { _id: ObjectId; }>>>'.
      Types of property '$locals' are incompatible.
        Type 'Record<string, unknown>' is not assignable to type 'DeepPartial<Record<string, unknown>>'.
          'string' index signatures are incompatible.
            Type 'unknown' is not assignable to type 'DeepPartial<unknown> | undefined'.
              Type 'unknown' is not assignable to type 'DeepPartial<unknown>'.ts(2345)
```

Reproduction repository & branch: https://github.com/typegoose/typegoose-testing/tree/mongooseUpdateQueryDeepPartialUnknown

Steps:
- clone repository & checkout the mentioned branch (`mongooseUpdateQueryDeepPartialUnknown`)
- Checkout commit `mongoose 6.0.14` (hash `5ca376704e014c8f956e5f49d8fc16c8cc214f5a`) for behavior on mongoose 6.0.14
- run `yarn install`
- run `yarn run build` (result should not fail)
- Checkout commit `mongoose 6.0.15` (hash `e696fe7c5082866358f551d241f4489319a577ad`) for behavior on mongoose 6.0.15
- run `yarn install`
- run `yarn run build` and observe the typescript error
- Checkout commit `mongoose 6.1.1` (hash `b26dff3fddd5e0300b4b2c3f513abe13a79b253e`) for behavior on mongoose 6.1.1
- run `yarn install`
- run `yarn run build` and observe the typescript error (still)

Note: replace `yarn` with whatever package manager is used

tsconfig can be found in the mentioned branch and commit

**What is the expected behavior?**
No Error to happen

**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**
Started happening in 6.0.15
still exists in 6.1.1

---

PS: yes i know i am doing a `updateOne` operation on the same document with the same values, it is just that the easiest to show the types that way

**Comment by magyarb** [comment]
This is almost the same as #11090 , #11099 . Looks like there is a similar issue at update and find queries too.

**Comment by DavideViolante** [comment]
I tested your repo and it seems solved since >=6.1.2. Test with it or 6.1.3 and let us know.

**Comment by hasezoey** [comment]
can also confirm that 6.1.2 has fixed the problem can also confirm that it also does not happen in 6.1.3

closing because fixed since 6.1.2
